### PR TITLE
Hive refactors

### PR DESCRIPTION
**Pull request by sopel39**

### CODE REVIEW

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftMetastoreUtil.java [hunk 1]
```diff
@@ -499,7 +500,7 @@ public static Partition fromMetastoreApiPartition(org.apache.hadoop.hive.metasto
                 .setValues(partition.getValues())
                 .setColumns(schema.stream()
                         .map(ThriftMetastoreUtil::fromMetastoreApiFieldSchema)
-                        .collect(toList()))
+                        .collect(toImmutableList()))
```
Comment (findepi) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftMetastoreUtil.java]: there are 38 occurrences of `collect(toList()))` in presto-hive

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftMetastoreUtil.java [hunk 2]
```diff
@@ -660,7 +661,7 @@ private static long fromMetastoreDistinctValuesCount(long distinctValuesCount, l
 
     public static Set<RoleGrant> fromRolePrincipalGrants(Collection<RolePrincipalGrant> grants)
     {
-        return ImmutableSet.copyOf(grants.stream().map(ThriftMetastoreUtil::fromRolePrincipalGrant).collect(toList()));
+        return ImmutableSet.copyOf(grants.stream().map(ThriftMetastoreUtil::fromRolePrincipalGrant).collect(toImmutableList()));
```
Comment (findepi) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftMetastoreUtil.java]: `.collect(toImmutableSet())` would be more appropriate

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/SemiTransactionalHiveMetastore.java [hunk 3]
```diff
@@ -849,9 +848,13 @@ public synchronized void revokeRoles(Set<String> roles, Set<HivePrincipal> grant
 
     private String getTableOwner(HiveIdentity identity, String databaseName, String tableName)
     {
-        Table table = delegate.getTable(identity, databaseName, tableName)
+        return getSourceTableOrElseThrow(identity, databaseName, tableName).getOwner();
+    }
+
+    private Table getSourceTableOrElseThrow(HiveIdentity identity, String databaseName, String tableName)
```
Comment (findepi) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/SemiTransactionalHiveMetastore.java]: It's obvious that a method throws when it cannot return a value, so `OrElseThrow` doesn't add value.

```suggestion
    private Table getTable(HiveIdentity identity, String databaseName, String tableName)
```

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueHiveMetastore.java [hunk 4]
```diff
@@ -178,15 +178,15 @@ private static AWSCredentialsProvider getAwsCredentialsProvider(GlueHiveMetastor
     {
         if (config.getAwsAccessKey().isPresent() && config.getAwsSecretKey().isPresent()) {
             return new AWSStaticCredentialsProvider(
-                new BasicAWSCredentials(config.getAwsAccessKey().get(), config.getAwsSecretKey().get()));
+                    new BasicAWSCredentials(config.getAwsAccessKey().get(), config.getAwsSecretKey().get()));
```
Comment (findepi) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueHiveMetastore.java]: Please also fix indentation in `presto-hive/src/main/java/io/prestosql/plugin/hive/ConnectorObjectNameGeneratorModule.java`, `presto-hive/src/main/java/io/prestosql/plugin/hive/HivePartitionManager.java` and  `presto-hive/src/test/java/io/prestosql/plugin/hive/TestBackgroundHiveSplitLoader.java`

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftMetastoreUtil.java (hunk 2 above)
Comment (kokosing) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftMetastoreUtil.java]: No no need to copy then. Also you can fix formatting (not to use single line chained trains).

File: presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/SemiTransactionalHiveMetastore.java (hunk 3 above)
Comment (findepi) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/SemiTransactionalHiveMetastore.java]: 😞

Comment (sopel39) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/SemiTransactionalHiveMetastore.java]: `getTable` doesn't work as it's conflicting with other `getTable`. Maybe `getSourceTable` is better: https://github.com/prestosql/presto/pull/1880

**Pull request merged**